### PR TITLE
First commit, add cache option to zeep and create a basic test case w…

### PR DESCRIPTION
**Pull request by TristanFAURE**
#140 
First commit, add cache option to zeep and create a basic test case with no properties

also add skipped status 

Work in Progress, only for a first status and first comments

### CODE REVIEW

**Comment by jesper-raemaekers** [comment]
Hi,

Just to let you know, the test runner in DevOps runs on Polarion 21r2 which has no tokens to sign in. I chedk and your tests work fine on it when using the password, so please use that in your final commit.

I'm curious.. does the cache improve performance a lot?

Thanks for helping out!

**Comment by TristanFAURE** [comment]
Hi

> Hi,
> 
> Just to let you know, the test runner in DevOps runs on Polarion 21r2 which has no tokens to sign in. I chedk and your tests work fine on it when using the password, so please use that in your final commit.

ok no problem, thanks for the information ! By the way with the token authentication there is a performance issue if no user is provided (data cannot retrieved from services dict). So in my tests I put "_" in username or something like that

> 
> I'm curious.. does the cache improve performance a lot?
> 
> Thanks for helping out!

It should I wanted to ask somebody to test as I have few existing metrics ... I also need somebody to test on a windows environment. The cache is used to store wsdl metadata

Another change that should improve performance is 
```python
if self.transport is None:
```
because Session was created a lot of time

By the way do you have feedbacks about our suggestion to add traceability links thanks to properties ?

File: polarion/polarion.py [hunk 1]
```diff
@@ -2,8 +2,13 @@
 import re
 from urllib.parse import urljoin, urlparse
 import requests
+import tempfile
+import os
 from zeep import Client, Transport
 from zeep.plugins import HistoryPlugin
+from zeep import Client
```
Comment (LionnelC) [inline polarion/polarion.py:9]: Duplicates
```suggestion
```

**Review by LionnelC — COMMENTED** [review]
Sorry, not have many times to check deeper

**Comment by TristanFAURE** [comment]
@jesper-raemaekers 
it seems on CI there is an issue with file opening. I put './tests/junit/no_properties.xml' how are run the tests on the CI (in which path)? how I have to change the path to comply ?

**Comment by TristanFAURE** [comment]
@jesper-raemaekers it seems that errors are in tests because of fields not available on your test server instance. Can you share with me the fields and traceability links configured ? (or add mine in your instance)

**Comment by TristanFAURE** [comment]
we will see the result of the tests, but in terms of features it is ok for #140 

If you are ok with the pull request I will update the doc to explain the xml import

**Comment by TristanFAURE** [comment]
For the future I think we will add possiblity to create workitems from test suites we need kind of mapping information, but it will be another issue, so you can review this PR

**Comment by TristanFAURE** [comment]
> Hi,
> 
> Just to let you know, the test runner in DevOps runs on Polarion 21r2 which has no tokens to sign in. I chedk and your tests work fine on it when using the password, so please use that in your final commit.
> 
> I'm curious.. does the cache improve performance a lot?
> 
> Thanks for helping out!

And about the cache for a process with a duration of 38 sec with the cache it is now 30 sec

**Comment by TristanFAURE** [comment]
@LionnelC @jesper-raemaekers 
who will perform the review ? please tell me if some changes are needed

**Comment by jesper-raemaekers** [comment]
> @LionnelC @jesper-raemaekers who will perform the review ? please tell me if some changes are needed

Sorry, got to it now!



> > Hi,
> > Just to let you know, the test runner in DevOps runs on Polarion 21r2 which has no tokens to sign in. I chedk and your tests work fine on it when using the password, so please use that in your final commit.
> > I'm curious.. does the cache improve performance a lot?
> > Thanks for helping out!
> 
> And about the cache for a process with a duration of 38 sec with the cache it is now 30 sec

Nice

**Pull request merged**